images/linux/scripts/installers/codeql-bundle.sh

```diff
@@ -4,22 +4,37 @@
 ##  Desc:  Install the CodeQL CLI Bundle to the toolcache.
 ################################################################################
 
-source $HELPER_SCRIPTS/install.sh
+source "$HELPER_SCRIPTS/install.sh"
 
 # Retrieve the name of the CodeQL bundle preferred by the Action (in the format codeql-bundle-YYYYMMDD).
-codeql_bundle_name="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json | jq -r .bundleVersion)"
-# Convert the bundle name to a version number (0.0.0-YYYYMMDD).
-codeql_bundle_version="0.0.0-${codeql_bundle_name##*-}"
+defaults="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json)"
+codeql_bundle_name="$(${defaults} | jq -r .bundleVersion)"
+codeql_cli_version="$(${defaults} | jq -r .cliVersion)"
+prior_codeql_bundle_name="$(${defaults} | jq -r .priorBundleVersion)"
+prior_codeql_cli_version="$(${defaults} | jq -r .priorCliVersion)"
 
-extraction_directory="$AGENT_TOOLSDIRECTORY/CodeQL/$codeql_bundle_version/x64"
-mkdir -p "$extraction_directory"
+# Convert the bundle names to bundles with a version number (x.y.z-YYYYMMDD).
+codeql_bundle_version="${codeql_cli_version}-${codeql_bundle_name##*-}"
+prior_codeql_bundle_version="${prior_codeql_cli_version}-${prior_codeql_bundle_name##*-}"
 
-echo "Downloading CodeQL bundle $codeql_bundle_version..."
-download_with_retries "https://github.com/github/codeql-action/releases/download/$codeql_bundle_name/codeql-bundle.tar.gz" "/tmp" "codeql-bundle.tar.gz"
-tar -xzf "/tmp/codeql-bundle.tar.gz" -C "$extraction_directory"
+# For both of the bundle versions, download and name appropriately.
+for bundle in "${codeql_bundle_version} ${codeql_bundle_name}" "${prior_codeql_bundle_version} ${prior_codeql_bundle_name}"
+do
+    set -- "$bundle"
```

Comment:
This assumes that there are no spaces in `codeql_bundle_version` or `codeql_bundle_name`.  This should be a safe assumption.